torch_xla/csrc/xla_generator.cpp

```diff
@@ -5,10 +5,102 @@
 #include <ATen/core/Tensor.h>
 #include <c10/core/Device.h>
 #include <c10/core/DeviceType.h>
+#include <c10/core/GeneratorImpl.h>
 #include <c10/core/TensorImpl.h>
+#include <c10/util/CallOnce.h>
 #include <c10/util/intrusive_ptr.h>
 
+// XLA headers
 #include <cstring>
+#include <deque>
+#include <vector>
+
+#include "torch_xla/csrc/aten_xla_bridge.h"
+#include "torch_xla/csrc/runtime/computation_client.h"
+#include "torch_xla/csrc/runtime/runtime.h"
+
+namespace at {
+
+namespace detail {
+
+namespace {
+
+// Total number of XLA devices in the system.
+static int64_t num_xla_devices;
+
+// Ensures default_gens_xla is initialized once.
+static std::deque<c10::once_flag> xla_gens_init_flag;
+
+// Default, global XLA generators, one per XLA device.
+static std::vector<at::Generator> default_gens_xla;
+
+/*
+ * Populates the global variables related to XLA generators
+ * Warning: this function must only be called once!
+ */
+static void initXLAGenVector() {
+  // Ensures we only call deviceCount only once.
+  static bool num_xla_device_init_flag [[maybe_unused]] = []() {
+    // Get local num of XLA devices
+    auto maybe_client = torch_xla::runtime::GetComputationClient();
+    if (!maybe_client.ok()) {
+      // If runtime client initialization failed, default to 1 device
+      num_xla_devices = 1;
+    } else {
+      auto* client = maybe_client.value();
+      num_xla_devices = static_cast<int64_t>(client->GetNumDevices());
+    }
```

Comment:
I think it's better to error, whenever `GetComputationClient()` returns with an error status.
Check the macros in [`status.h`][1] for that.

[1]: https://github.com/pytorch/xla/blob/master/torch_xla/csrc/status.h